src/OpenColorIO/CMakeLists.txt

```diff
@@ -182,12 +182,18 @@ if(NOT WIN32)
     configure_file(res/OpenColorIO.pc.in ${CMAKE_CURRENT_BINARY_DIR}/OpenColorIO.pc @ONLY)
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/OpenColorIO.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 endif()
+
 add_library(OpenColorIO ${SOURCES})
 
+# Require at least a C++11 compatible compiler for consumer projects.
+target_compile_features(OpenColorIO
+    PUBLIC cxx_std_11
```

Comment:
This is enforced in the root CMake project via this module:
https://github.com/AcademySoftwareFoundation/OpenColorIO/blob/master/share/cmake/utils/CppVersion.cmake

Do you think that's sufficient? Or that having the requirement directly on the target is beneficial too?

Reply:
The main idea was to pass this requirement onto the project importing OpenColorIO. If you look at the generated `OpenColorIOTargets.cmake`, it should now include a `INTERFACE_COMPILE_FEATURES "cxx_std_11"` which was not the case before. That way client project should get a nice error if using an older standard. The `CppVersion.cmake` script only set the default CMake variables to be used to initialise subsequent targets declared within the project.

Does that that answer your question or did I misinterpret ?

Reply:
That's something important to enforce/impose at the API level because it's a strong requirement imposed by OpenColorIO to apps.

Reply:
Makes sense. Thanks.